core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreReaderThread.java

```diff
@@ -79,57 +71,49 @@ public KafkaStoreReaderThread(ZkUtils zkUtils,
+    consumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapBrokers);
+    consumerProps.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+    consumerProps.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+    consumerProps.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+            org.apache.kafka.common.serialization.ByteArrayDeserializer.class);
+    consumerProps.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+            org.apache.kafka.common.serialization.ByteArrayDeserializer.class);
+    this.consumer = new KafkaConsumer<>(consumerProps);
+
+    int partitionCount = this.consumer.partitionsFor(this.topic).size();
+    if (partitionCount < 1) {
       throw new IllegalArgumentException("Unable to subscribe to the Kafka topic " + topic +
                                          " backing this data store. Topic may not exist.");
+    } else if (partitionCount > 1) {
+      throw new IllegalStateException("Unexpected number of partitions in the " + topic +
+                                      " topic. Expected 1 and instead got " + partitionCount);
     }
-    KafkaStream<byte[], byte[]> stream = streamsForTheLogTopic.get(0);
-    consumerIterator = stream.iterator();
+
+    this.topicPartition = new TopicPartition(topic, 0);
+    this.consumer.assign(Arrays.asList(this.topicPartition));
+    this.consumer.seekToBeginning(this.topicPartition);
+
+    log.info("Initialized last consumed offset to " + offsetInSchemasTopic);
+
     log.debug("Kafka store reader thread started with consumer properties " +
               consumerProps.toString());
   }
 
-  /**
-   * Fetch the offset of the last consumed message from ZK.
-   */
-  private long offsetOfLastConsumedMessage(ZkUtils zkUtils, String group, String topic) {
-    Option<String> committedOffsetStringOpt = zkUtils.readDataMaybeNull(
-        String.format("/consumers/%s/offsets/%s/0", group, topic))._1();
-    if (committedOffsetStringOpt.isEmpty()) {
-      return -1L;
-    } else {
-      // the offset of the last consumed message is always one less than the last committed offset
-      return Long.parseLong(committedOffsetStringOpt.get()) - 1;
-    }
-  }
-
   @Override
   public void doWork() {
     try {
-      if (consumerIterator.hasNext()) {
-        MessageAndMetadata<byte[], byte[]> messageAndMetadata = consumerIterator.next();
-        byte[] messageBytes = messageAndMetadata.message();
+      // don't need to commit offsets because the log is always read from the beginning on startup
+      ConsumerRecords<byte[], byte[]> records = consumer.poll(Long.MAX_VALUE);
+      for (ConsumerRecord<byte[], byte[]> record : records) {
         K messageKey = null;
         try {
-          messageKey = this.serializer.deserializeKey(messageAndMetadata.key());
+          messageKey = this.serializer.deserializeKey(record.key());
         } catch (SerializationException e) {
           log.error("Failed to deserialize the schema or config key", e);
```

Comment:
Similar to the comment below. If we ignore this exception, I think we'd get an NPE on line 121.